### PR TITLE
Changed "Cancel" to "Close" in some dialogs

### LINKED ISSUE #1781: Some dialogs use `Cancel` instead of `Close`
Some dialogs use „Cancel“ instead of „Close“ to close the dialog, e.g. user status or settings. We should think about using „Close“ as we don’t actually cancel anything here

### PR DESCRIPTION
**Pull request by rapterjet2004**
- fixes #1781 

Hopefully got them all

### CODE REVIEW

File: NextcloudTalk/QRCodeLoginController.swift [hunk 1]
```diff
@@ -72,7 +72,7 @@ import QRCodeReader
                     }
                 }))
 
-                alert.addAction(UIAlertAction(title: NSLocalizedString("Cancel", comment: ""), style: .cancel, handler: nil))
+                alert.addAction(UIAlertAction(title: NSLocalizedString("Close", comment: ""), style: .cancel, handler: nil))
```
Comment (Ivansss) [inline NextcloudTalk/QRCodeLoginController.swift]: I would use "OK" for error dialogs when no further action from the user is needed (as we do in other error dialogs in this app).
In this particular case, we could even think of adding a "Settings" button to redirect the user into the app's settings as we do in:
https://github.com/nextcloud/talk-ios/blob/d7eb5acf2c378c94c1311598b22c3389b23ee104/NextcloudTalk/VoiceMessageTranscribeViewController.m#L172

File: NextcloudTalk/PollCreationViewController.swift [hunk 2]
```diff
@@ -87,7 +87,7 @@ import UIKit
         let alert = UIAlertController(title: NSLocalizedString("Creating poll failed", comment: ""),
                                       message: NSLocalizedString("An error occurred while creating the poll", comment: ""),
                                       preferredStyle: .alert)
-        alert.addAction(UIAlertAction(title: NSLocalizedString("Cancel", comment: ""), style: .cancel, handler: nil))
+        alert.addAction(UIAlertAction(title: NSLocalizedString("Close", comment: ""), style: .cancel, handler: nil))
```
Comment (Ivansss) [inline NextcloudTalk/PollCreationViewController.swift]: I would use "OK" for error dialogs when no further action from the user is needed (as we do in other error dialogs in this app).

File: NextcloudTalk/SettingsTableViewController.swift [hunk 3]
```diff
@@ -389,7 +389,7 @@ class SettingsTableViewController: UITableViewController, UITextFieldDelegate, U
         }
         failedPhoneNumberDialog.addAction(retryAction)
 
-        let cancelAction = UIAlertAction(title: NSLocalizedString("Cancel", comment: ""), style: .default, handler: nil)
+        let cancelAction = UIAlertAction(title: NSLocalizedString("Close", comment: ""), style: .default, handler: nil)
```
Comment (Ivansss) [inline NextcloudTalk/SettingsTableViewController.swift]: I would use "OK" for error dialogs when no further action from the user is needed (as we do in other error dialogs in this app).

File: NextcloudTalk/ShareLocationViewController.m [hunk 4]
```diff
@@ -150,7 +150,7 @@ - (void)showAuthorizationStatusDeniedAlert
         [[UIApplication sharedApplication] openURL:[NSURL URLWithString:UIApplicationOpenSettingsURLString] options:@{} completionHandler:nil];
     }];
     [alert addAction:settingsButton];
-    [alert addAction:[UIAlertAction actionWithTitle:NSLocalizedString(@"Cancel", nil) style:UIAlertActionStyleCancel handler:nil]];
+    [alert addAction:[UIAlertAction actionWithTitle:NSLocalizedString(@"Close", nil) style:UIAlertActionStyleCancel handler:nil]];
```
Comment (Ivansss) [inline NextcloudTalk/ShareLocationViewController.m]: I would use "OK" for error dialogs when no further action from the user is needed (as we do in other error dialogs in this app).
In this particular case, we could even think of adding a "Settings" button to redirect the user into the app's settings as we do in:
https://github.com/nextcloud/talk-ios/blob/d7eb5acf2c378c94c1311598b22c3389b23ee104/NextcloudTalk/VoiceMessageTranscribeViewController.m#L172

File: NextcloudTalk/VoiceMessageTranscribeViewController.m [hunk 5]
```diff
@@ -172,7 +172,7 @@ - (void)showSpeechRecognitionNotAvailable
         [[UIApplication sharedApplication] openURL:[NSURL URLWithString:UIApplicationOpenSettingsURLString] options:@{} completionHandler:nil];
     }];
     [alert addAction:settingsButton];
-    [alert addAction:[UIAlertAction actionWithTitle:NSLocalizedString(@"Cancel", nil)
+    [alert addAction:[UIAlertAction actionWithTitle:NSLocalizedString(@"Close", nil)
```
Comment (Ivansss) [inline NextcloudTalk/VoiceMessageTranscribeViewController.m]: I would use "OK" for error dialogs when no further action from the user is needed (as we do in other error dialogs in this app).

**Review by Ivansss — CHANGES_REQUESTED** [review]

**Comment by SystemKeeper** [comment]
Rebased because of deprecated CI action used. Should run now.

**Review by Ivansss — APPROVED** [review]
Thanks a lot for this PR @rapterjet2004 :)

**Pull request merged**